apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/api/CompilerParams.kt

```diff
@@ -15,75 +15,105 @@ import org.gradle.api.provider.Provider
  */
 interface CompilerParams {
   /**
-   * Whether to generate java (default) or kotlin models
+   * Whether to generate Java or Kotlin models
+   *
+   * Default value: false
    */
   val generateKotlinModels: Property<Boolean>
 
   /**
-   * Whether to generate the operation output json. The json contains information such as
-   * operation id, name and complete source sent to the server. This can be useful if you need to upload
+   * Whether to generate OperationOutput.json. OperationOutput.json contains information such as
+   * operation id, name and complete source sent to the server. This can be used to upload
    * a query's exact content to a server that doesn't support automatic persisted queries.
    *
    * The operation output is written in [CompilationUnit.operationOutputFile]
+   *
+   * Default value: false
    */
   val generateOperationOutput: Property<Boolean>
 
 
   /**
    * For custom scalar types like Date, map from the GraphQL type to the jvm/kotlin type.
    *
-   * empty by default.
+   * Default value: the empty map
    */
   val customTypeMapping: MapProperty<String, String>
 
   /**
-   * For custom persisted query Ids.
+   * By default, Apollo uses `Sha256` hashing algorithm to generate an ID for the query.
+   * To provide a custom ID generation logic, pass an `instance` that implements the [OperationIdGenerator]. How the ID is generated is
+   * indifferent to the compiler. It can be an hashing algorithm or generated by a backend.
+   *
+   * Example Md5 hash generator:
+   * ```groovy
+   * import com.apollographql.apollo.compiler.OperationIdGenerator
+   *
+   * apollo {
+   *   operationIdGenerator = new OperationIdGenerator() {
+   *     String apply(String operationDocument, String operationFilepath) {
+   *       return operationDocument.md5()
+   *     }
+   *
+   *     /**
+   *      * Use this version override to indicate an update to the implementation.
+   *      * This invalidates the current cache.
+   *      */
+   *     String version = "v1"
+   *   }
+   * }
+   * ```
    *
-   * If not provided, default hashing algorithm (sha256) will be used
+   * Default value: [OperationIdGenerator.Sha256]
    */
   val operationIdGenerator: Property<OperationIdGenerator>
 
   /**
    * The custom types code generate some warnings that might make the build fail.
    * suppressRawTypesWarning will add the appropriate SuppressWarning annotation
    *
-   * false by default
+   * Default value: false
    */
   val suppressRawTypesWarning: Property<Boolean>
 
   /**
-   * Whether to suffix your queries, etc.. with `Query`, etc..
+   * When true, Apollo-Android will make sure all the generated classes end with 'Query' or 'Mutation'.
```

Comment:
Small: I know the whole project is still called Apollo Android but it would be better to avoid Android in docs if it is not Android specific. What about just saying Apollo? Or you can go passive and say "all generated classes will end with ..."